### PR TITLE
[GUILD] - Convert Menu to typescript

### PR DESCRIPTION
**Pull request by eniskraasniqi1**
### Description

This PR converts Menu component to typescript.

**Note**: 

Menu.js might need more strict checking, since I tried to narrow down all old lifecycle methods into 2 seperate useEffects, just in case I missed something :).

### CODE REVIEW

**Review by JorenSaeyTL — DISMISSED** [review]

File: src/components/menu/Menu.tsx [hunk 1]
```diff
@@ -0,0 +1,237 @@
...
+      onSelect(value);
+    }
+
+    if (onClick) {
+      event.persist();
+      onClick(event);
+    }
+
+    // eslint-disable-next-line @typescript-eslint/no-use-before-define
+    hide();
```
Comment (qubis741) [inline src/components/menu/Menu.tsx:99]: why you dont move it up so its lint-okay? same for other occurrences

File: src/components/menu/menu.stories.tsx [hunk 2]
```diff
@@ -11,7 +12,7 @@ export default {
   title: addStoryInGroup(MID_LEVEL_BLOCKS, 'Menu'),
 };
```
Comment (qubis741) [inline src/components/menu/menu.stories.tsx:13]: add here as `ComponentMeta<typeof Menu >;`

File: src/components/menu/Menu.tsx (hunk 1 above)
Comment (eniskraasniqi1) [reply src/components/menu/Menu.tsx:99]: Well that's the only simpliest way I found, there are chained functions. If you move `hide` up, then you have to move `removeEvents` as well. And then `removeEvents` has `handleDocumentClick` which calls `hide`. 😄

File: src/components/menu/iconMenu.stories.tsx [hunk 3]
```diff
@@ -1,13 +1,14 @@
 import React from 'react';
 import { addStoryInGroup, MID_LEVEL_BLOCKS } from '../../../.storybook/utils';
 import { IconMenu, MenuItem, MenuDivider } from '../../index';
+import { ComponentStory } from '@storybook/react';
 
 export default {
   component: IconMenu,
   title: addStoryInGroup(MID_LEVEL_BLOCKS, 'IconMenu'),
 };
```
[the remark this answers is not in the document]
Comment (eniskraasniqi1) [reply src/components/menu/iconMenu.stories.tsx]: Aaah, so it infers the props from the component itself. Didn't know this, thank you 🙏

**Review by qubis741 — APPROVED** [review]

**Comment by qubis741** [comment]
Manually checked :white_check_mark:

File: src/components/menu/Menu.tsx [hunk 4]
```diff
@@ -0,0 +1,237 @@
...
+import { events } from '../utils';
+import { getViewport } from '../utils/utils';
+import Box, { pickBoxProps } from '../box';
+import MenuItem from './MenuItem';
+import theme from './theme.css';
+import uiUtilities from '@teamleader/ui-utilities';
+import { BoxProps } from '../box/Box';
+import { GenericComponent } from '../../@types/types';
+
+export type POSITIONS = 'auto' | 'static' | 'top-left' | 'top-right' | 'bottom-left' | 'bottom-right';
```
Comment (BeirlaenAaron) [inline src/components/menu/Menu.tsx:12]: PP: types should be PascalCase

**Review by BeirlaenAaron — APPROVED** [review]

File: src/components/menu/Menu.tsx (hunk 4 above)
Comment (eniskraasniqi1) [reply src/components/menu/Menu.tsx:12]: Yeah, good point, will change it so we have consistency 😉. Thanks!

**Review by BeirlaenAaron — APPROVED** [review]

**Review by stefaandevylder — APPROVED** [review]

**Comment by stefaandevylder** [comment]
MC'ing...

**Review by jelledc — APPROVED** [review]

**Pull request merged**